src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -120,6 +152,14 @@ private void Include(Predicate<Instrument> instrumentFilter)
             }
         }
 
+        public void SetHistogramAggregation(HistogramAggregation histogramAggregation)
+        {
+            lock (this)
+            {
+                _histogramAggregation = histogramAggregation;
+            }
```

Comment:
Just curious what this lock is for? Aggregation is a class so this write should be atomic. I guess it is here to prevent switching aggregation while some process of stats collection is ongoing or something like that? If so then maybe a quick comment would help.

Reply:
Thanks!

I have added the comment. Let me know if you have any more comments.